src/main/java/net/fabricmc/meta/utils/PomParser.java

```diff
@@ -49,18 +50,28 @@ private void load() throws IOException, XMLStreamException {
 		versions.clear();
 
 		URL url = new URL(path);
-		XMLStreamReader reader = XMLInputFactory.newInstance().createXMLStreamReader(url.openStream());
-		while (reader.hasNext()) {
-			if (reader.next() == XMLStreamConstants.START_ELEMENT && reader.getLocalName().equals("version")) {
-				String text = reader.getElementText();
-				versions.add(text);
+		URLConnection conn = url.openConnection();
+		conn.setConnectTimeout(3000);
+		conn.setReadTimeout(3000);
+		XMLStreamReader reader = null;
+
+		try {
+			reader = XMLInputFactory.newInstance().createXMLStreamReader(conn.getInputStream());
```

Comment:
Why not use try with resources?

Reply:
I tried that first, but it isn't Closeable -_-